src/Chirp.Infrastructure/Repositories/CheepRepository.cs

```diff
@@ -4,16 +4,30 @@
 namespace Chirp.Infrastructure;
 
 public class CheepRepository : ICheepRepository {
-    
-    public Task<List<CheepDto>> GetCheeps(int page, string? author = null) {
-        using var context = new ChirpContext();
+    private readonly ChirpContext _dbContext;
+
+    public CheepRepository(ChirpContext dbContext) =>
+        _dbContext = dbContext;
 
-        return context.Cheeps
-            .Where( cheep => cheep.Author.Name == author || author == null)
+    public Task<List<CheepDto>> GetCheeps(int page, string? author = null) {
+        return _dbContext.Cheeps
+            .Where(cheep => cheep.Author.Name == author || author == null)
             .OrderByDescending(cheep => cheep.TimeStamp)
-            .Skip( 32 * (page - 1))
+            .Skip(32 * (page - 1))
             .Take(32)
-            .Select( cheep => new CheepDto(cheep.Author.Name, cheep.Text, cheep.TimeStamp.ToString("MM/dd/yy H:mm:ss")))
+            .Select(cheep => new CheepDto(cheep.Author.Name, cheep.Text, cheep.TimeStamp.ToString("MM/dd/yy H:mm:ss")))
             .ToListAsync();
     }
+
+    public async Task AddCheep(string message, string authorName) {
+        // We have no way of knowing the correct email if author does not exist already
+        var author = _dbContext.Authors.Where(author => author.Name == authorName).FirstOrDefault(new Author() {
+            Id = Guid.NewGuid(),
+            Name = authorName,
+            Email = "thisisalegitemail@email.email",
+            Cheeps = new List<Cheep>()
+        });
+
+        await _dbContext.Cheeps.AddAsync(new Cheep() { Id = Guid.NewGuid(), AuthorId = author.Id, Author = author, Text = message });
```

Comment:
As commented on other code that add an entry in the database. There should probably be a call to _context.SaveChanges().